### PR TITLE
OADP 3400 storage class mapping

### PR DESCRIPTION
**Pull request by CarmiWisemon**
OADP 1.3.1

OCP 4.13+

Resolves - https://issues.redhat.com/browse/OADP-3400

Deploy preview - https://72284--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-advanced-topics#oadp-storage-class-mapping

### CODE REVIEW

File: modules/oadp-storage-class-mapping.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+You can use the `change-storage-class-config` field to change the storage class of your data objects. This feature allows you to optimize costs and performance by moving data between different storage tiers, such as from standard to archival storage, based on your needs and access patterns.
+
+[id=storage-class-mapping-mtc_{context}]
+== Storage class mapping with MTC
+
+You can use Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one OpenShift Container Platform cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the Migration Toolkit for Containers (MTC) web console.
+
+For more information see link:https://docs.openshift.com/container-platform/4.15/migration_toolkit_for_containers/advanced-migration-options-mtc.html#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes]
```
Comment (anarnold97) [inline modules/oadp-storage-class-mapping.adoc:18]:
```suggestion
For more information see xref:../../backup_and_restore/advanced-migration-options-mtc.adoc#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes].
```

Comment (CarmiWisemon) [reply modules/oadp-storage-class-mapping.adoc]: @anarnold97 
I tried replacing the link with the xref but it didn't work and it broke the build.

Comment (anarnold97) [reply modules/oadp-storage-class-mapping.adoc]: Well, fix the xref

File: modules/oadp-storage-class-mapping.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+You can use the `change-storage-class-config` field to change the storage class of your data objects. This feature allows you to optimize costs and performance by moving data between different storage tiers, such as from standard to archival storage, based on your needs and access patterns.
+
+[id=storage-class-mapping-mtc_{context}]
+== Storage class mapping with MTC
+
+You can use Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one OpenShift Container Platform cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the Migration Toolkit for Containers (MTC) web console.
+
+For more information see link:https://docs.openshift.com/container-platform/4.15/migration_toolkit_for_containers/advanced-migration-options-mtc.html#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes].
```
Comment (anarnold97) [inline modules/oadp-storage-class-mapping.adoc:18]:
```suggestion
For more information see xref:../../migration_toolkit_for_containers/advanced-migration-options-mtc.adoc#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes].
```

Comment (CarmiWisemon) [reply modules/oadp-storage-class-mapping.adoc]: @anarnold97 Thank you.!!! This worked.

File: modules/oadp-storage-class-mapping.adoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+== Storage class mapping with MTC
+
+You can use Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one OpenShift Container Platform cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the Migration Toolkit for Containers (MTC) web console.
+
+For more information see xref:../../migration_toolkit_for_containers/advanced-migration-options-mtc.adoc#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes].
+
+[id=storage-class-mapping-oadp_{context}]
+== Storage class mapping with OADP
+
+You can also use OpenShift API for Data Protection (OADP)  with the Velero plugin for vsphere v1.1.0 and above to change the storage class of a persistent volume (PV) during restores, by configuring a storage class mapping in the config map in the Velero namespace.
```
Comment (stillalearner) [inline modules/oadp-storage-class-mapping.adoc:23]:
```suggestion
You can also use OpenShift API for Data Protection (OADP)  with the Velero plugin v1.1.0 and above to change the storage class of a persistent volume (PV) during restores, by configuring a storage class mapping in the config map in the Velero namespace.
```

File: modules/oadp-storage-class-mapping.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+[id="oadp-storage-class-mapping_{context}"]
+:_mod-docs-content-type: CONCEPT
+= Storage class mapping
+
+Storage class mapping allows you to define rules or policies specifying which storage class should be applied to different types of data. This feature automates the process of determining storage classes based on access frequency, data importance, and cost considerations. It optimizes storage efficiency and cost-effectiveness by ensuring that data is stored in the most suitable storage class for its characteristics and usage patterns.
+
+You can use the `change-storage-class-config` field to change the storage class of your data objects. This feature allows you to optimize costs and performance by moving data between different storage tiers, such as from standard to archival storage, based on your needs and access patterns.
+
+[id=storage-class-mapping-mtc_{context}]
+== Storage class mapping with MTC
```
Comment (mjpytlak) [inline modules/oadp-storage-class-mapping.adoc:14]: Should this be an H3? Seems like a type of Storage Mapping.

File: modules/oadp-storage-class-mapping.adoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+You can use Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one OpenShift Container Platform cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the Migration Toolkit for Containers (MTC) web console.
```
Comment (mjpytlak) [inline modules/oadp-storage-class-mapping.adoc:16]:
```suggestion
You can use the Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one {product-title} cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the Migration Toolkit for Containers (MTC) web console.
```

File: modules/oadp-storage-class-mapping.adoc [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+For more information see xref:../../migration_toolkit_for_containers/advanced-migration-options-mtc.adoc#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes].
+
+[id=storage-class-mapping-oadp_{context}]
+== Storage class mapping with OADP
```
Comment (mjpytlak) [inline modules/oadp-storage-class-mapping.adoc:21]: Should this be an H3? Seems like a type of Storage Mapping.

File: modules/oadp-storage-class-mapping.adoc [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+You can use the `change-storage-class-config` field to change the storage class of your data objects. This feature allows you to optimize costs and performance by moving data between different storage tiers, such as from standard to archival storage, based on your needs and access patterns.
+
+[id=storage-class-mapping-mtc_{context}]
+== Storage class mapping with MTC
+
+You can use Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one OpenShift Container Platform cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the Migration Toolkit for Containers (MTC) web console.
+
+For more information see xref:../../migration_toolkit_for_containers/advanced-migration-options-mtc.adoc#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes].
```
Comment (mjpytlak) [inline modules/oadp-storage-class-mapping.adoc:18]: Unfortunately, you cannot include an xref in a module.I suggest some along the lines of, For more information on XYZ, see "Additional" resources and add the link there.

File: modules/oadp-storage-class-mapping.adoc [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+[id=storage-class-mapping-oadp_{context}]
+== Storage class mapping with OADP
+
+You can also use OpenShift API for Data Protection (OADP)  with the Velero plugin v1.1.0 and above to change the storage class of a persistent volume (PV) during restores, by configuring a storage class mapping in the config map in the Velero namespace.
```
Comment (mjpytlak) [inline modules/oadp-storage-class-mapping.adoc:23]:
```suggestion
You can use OpenShift API for Data Protection (OADP) with the Velero plugin v1.1.0 and above to change the storage class of a persistent volume (PV) during restores by configuring a storage class mapping in the config map in the Velero namespace.
```

File: modules/oadp-storage-class-mapping.adoc [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+To employ ConfigMap with OADP, use the `change-storage-class-config` field. Ensure to change the storage class mapping based on your cloud provider.
```
Comment (mjpytlak) [inline modules/oadp-storage-class-mapping.adoc:25]: I suggest changing employ to "use" or "deploy".

File: modules/oadp-storage-class-mapping.adoc [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+. Change the storage class mapping by running the following command:
```
Comment (mjpytlak) [inline modules/oadp-storage-class-mapping.adoc:27]: You have a procedure inside of a procedure module. I suggest breaking this content out into its own module.

**Review by mjpytlak — COMMENTED** [review]
@CarmiWisemon Overall good work. Some suggestions. Some items that need to be addressed ahead of a merge. Thank you.

File: backup_and_restore/application_backup_and_restore/oadp-advanced-topics.adoc [hunk 11]
```diff
@@ -25,6 +25,11 @@ include::modules/oadp-backing-up-opt-out.adoc[leveloffset=+3]
 include::modules/oadp-cluster-to-cluster-uid-and-gid-ranges.adoc[leveloffset=+2]
 include::modules/oadp-backing-and-restoring-from-cluster-to-cluster.adoc[leveloffset=+2]
 
+[id="oadp-storage-class-mapping"]
+== OADP storage class mapping
```
[the remark this answers is not in the document]
Comment (CarmiWisemon) [reply backup_and_restore/application_backup_and_restore/oadp-advanced-topics.adoc:29]: @mjpytlak 
It's also important that the assembly should be unified.
I can easily add an introductory sentence, but I prefer that this module seems congruent to the rest of the assembly. Can this be addressed at a later date?

File: modules/oadp-storage-class-mapping.adoc (hunk 7 above)
Comment (CarmiWisemon) [reply modules/oadp-storage-class-mapping.adoc:18]: Done

File: modules/oadp-storage-class-mapping.adoc (hunk 4 above)
Comment (CarmiWisemon) [reply modules/oadp-storage-class-mapping.adoc]: It's presently a level 3. When I made the change you requested I received an error message.

File: modules/oadp-storage-class-mapping.adoc (hunk 6 above)
Comment (CarmiWisemon) [reply modules/oadp-storage-class-mapping.adoc]: It's presently a level 3. When I made the change you requested I received an error message.

Comment (mjpytlak) [reply modules/oadp-storage-class-mapping.adoc:19]: Oh. You are right. Sorry about that. The headings are OK as is.

File: backup_and_restore/application_backup_and_restore/oadp-advanced-topics.adoc [hunk 12]
```diff
@@ -35,4 +41,6 @@ For more information about OADP Data Mover, see xref:../../backup_and_restore/ap
 
 For more information about using Restic with OADP, see xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#backing-up-applications[Backing up applications with File System Backup: Kopia or Restic].
 
+For more information about MTC converting storage classes see xref:../../migration_toolkit_for_containers/advanced-migration-options-mtc.adoc#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes].
```
Comment (jeana-redhat) [inline backup_and_restore/application_backup_and_restore/oadp-advanced-topics.adoc:44]: Similar to Mike's comment above, this is not compliant with Additional resources formatting but I see it's based on existing content. This one is a pretty easy fix though, and directly impacts how things render on the portal; I think it's worth fixing before merge.

To make this compliant, convert these four items to a bulleted list, and remove all text that's not in the link text, like so:

```
* xref:../../backup_and_restore/application_backup_and_restore/oadp-advanced-topics.adoc#oadp-different-kubernetes-api-versions[Working with different Kubernetes API versions on the same cluster]
* xref:../../backup_and_restore/application_backup_and_restore/installing/oadp-using-data-mover-for-csi-snapshots-doc.adoc#backing-up-applications[Using Data Mover for CSI snapshots]
* xref:../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#backing-up-applications[Backing up applications with File System Backup: Kopia or Restic]
* xref:../../migration_toolkit_for_containers/advanced-migration-options-mtc.adoc#migration-converting-storage-classes_advanced-migration-options-mtc[Migration converting storage classes]
```

File: modules/oadp-storage-class-mapping.adoc [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+You can use the `change-storage-class-config` field to change the storage class of your data objects, which lets you optimize costs and performance by moving data between different storage tiers, such as from standard to archival storage, based on your needs and access patterns.
+
+[id=storage-class-mapping-mtc_{context}]
+== Storage class mapping with MTC
+
+You can use the Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one {product-title} cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the Migration Toolkit for Containers (MTC) web console.
```
Comment (jeana-redhat) [inline modules/oadp-storage-class-mapping.adoc:16]:
```suggestion
You can use the Migration Toolkit for Containers (MTC) to migrate containers, including application data, from one {product-title} cluster to another cluster and for storage class mapping and conversion. You can convert the storage class of a persistent volume (PV) by migrating it within the same cluster. To do so, you must create and run a migration plan in the MTC web console.
```

File: modules/oadp-storage-class-mapping-oadp.adoc [hunk 14]
```diff
@@ -0,0 +1,42 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/advanced-topics.adoc
+
+[id="oadp-storage-class-mapping-oadp_{context}"]
+:_mod-docs-content-type: PROCEDURE
+
+[id=storage-class-mapping-oadp_{context}]
+= Storage class mapping with OADP
```
Comment (jeana-redhat) [inline modules/oadp-storage-class-mapping-oadp.adoc:9]: Proc should have a gerund title, does this work..?
```suggestion
= Mapping storage classes with OADP
```

File: modules/oadp-storage-class-mapping-oadp.adoc [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+You can use OpenShift API for Data Protection (OADP)  with the Velero plugin v1.1.0 and above to change the storage class of a persistent volume (PV) during restores, by configuring a storage class mapping in the config map in the Velero namespace.
```
Comment (jeana-redhat) [inline modules/oadp-storage-class-mapping-oadp.adoc:11]: Per ISG
```suggestion
You can use OpenShift API for Data Protection (OADP)  with the Velero plugin v1.1.0 and later to change the storage class of a persistent volume (PV) during restores, by configuring a storage class mapping in the config map in the Velero namespace.
```

File: modules/oadp-storage-class-mapping-oadp.adoc [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+
+To deploy ConfigMap with OADP, use the `change-storage-class-config` field. Ensure to change the storage class mapping based on your cloud provider.
+
```
Comment (jeana-redhat) [inline modules/oadp-storage-class-mapping-oadp.adoc:14]:
```suggestion

.Procdure

```

File: modules/oadp-storage-class-mapping-oadp.adoc [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+[id="oadp-storage-class-mapping-oadp_{context}"]
+:_mod-docs-content-type: PROCEDURE
+
+[id=storage-class-mapping-oadp_{context}]
+= Storage class mapping with OADP
+
+You can use OpenShift API for Data Protection (OADP)  with the Velero plugin v1.1.0 and above to change the storage class of a persistent volume (PV) during restores, by configuring a storage class mapping in the config map in the Velero namespace.
+
+To deploy ConfigMap with OADP, use the `change-storage-class-config` field. Ensure to change the storage class mapping based on your cloud provider.
```
Comment (jeana-redhat) [inline modules/oadp-storage-class-mapping-oadp.adoc:13]:
```suggestion
To deploy ConfigMap with OADP, use the `change-storage-class-config` field. You must change the storage class mapping based on your cloud provider.
```

**Review by jeana-redhat — COMMENTED** [review]
Just a couple things to address here, this is rally nice though :)

/remove-label merge-review-in-progress
/remove-label merge-review-needed

File: backup_and_restore/application_backup_and_restore/oadp-advanced-topics.adoc (hunk 12 above)
Comment (CarmiWisemon) [reply backup_and_restore/application_backup_and_restore/oadp-advanced-topics.adoc:44]: @jeana-redhat 
Thank you for your suggestion.

File: modules/oadp-storage-class-mapping-oadp.adoc (hunk 14 above)
Comment (CarmiWisemon) [reply modules/oadp-storage-class-mapping-oadp.adoc:9]: Thank you!!!

**Comment by CarmiWisemon** [comment]
@jeana-redhat 
Thank you very much for your very helpful suggestions.

**Review by jeana-redhat — APPROVED** [review]
LGTM, thanks!

/remove-label merge-review-in-progress
/remove-label merge-review-needed

**Pull request merged**